variables.tf

```diff
@@ -1,264 +1,375 @@
-  type        = number
-  default     = null
+variable "subdomains" {
+  description = "An optional list of subdomains to use for API gateway"
+  type        = list(string)
+  default     = []
 }
 
-variable "default_stage_access_log_group_kms_key_id" {
-  description = "The ARN of the KMS Key to use when encrypting log data for Access logs"
-  type        = string
-  default     = null
+variable "subdomain_record_types" {
+  description = "A list of record types to create for the subdomain(s)"
+  type        = list(string)
+  default     = ["A", "AAAA"]
 }
 
-variable "default_stage_access_log_group_skip_destroy" {
-  description = "Set to true if you do not wish the log group (and any logs it may contain) to be deleted at destroy time, and instead just remove the log group from the Terraform state"
+################################################################################
+# Domain - Certificate
+################################################################################
+
+variable "create_certificate" {
+  description = "Whether to create a certificate for the domain"
   type        = bool
-  default     = false
+  default     = true
 }
 
-variable "default_stage_access_log_group_class" {
-  description = "Specified the log class of the Access log group. Possible values are: STANDARD or INFREQUENT_ACCESS"
-  type        = string
-  default     = null
-}
+################################################################################
+# Route(s) & Integration(s)
+################################################################################
 
-variable "default_stage_access_log_group_tags" {
-  description = "Additional tags for the Access logs"
-  type        = map(string)
-  default     = {}
+variable "create_routes_and_integrations" {
+  description = "Whether to create routes and integrations resources"
+  type        = bool
+  default     = true
 }
 
-#####
-# default stage API mapping
+variable "routes" {
+  description = "Map of API gateway routes with integrations"
+  type = map(object({
+    # Route
+    authorizer_key             = optional(string)
+    api_key_required           = optional(bool)
+    authorization_scopes       = optional(list(string), [])
+    authorization_type         = optional(string)
+    authorizer_id              = optional(string)
+    model_selection_expression = optional(string)
+    operation_name             = optional(string)
+    request_models             = optional(map(string), {})
+    request_parameter = optional(object({
+      request_parameter_key = optional(string)
+      required              = optional(bool, false)
+    }), {})
+    route_response_selection_expression = optional(string)
+
+    # Route settings
+    data_trace_enabled       = optional(bool, false)
+    detailed_metrics_enabled = optional(bool, false)
+    logging_level            = optional(string)
+    throttling_burst_limit   = optional(number, 500)
```

Comment:
yes, I would normally agree and normally not like to have to set arbitrary numbers either as a default value. However, when testing I kept getting errors only to discover that things were set to 0, and there are various upstream issues both in Terraform and other projects:
- https://github.com/hashicorp/terraform-provider-aws/issues/27674
- https://github.com/hashicorp/terraform-provider-aws/issues/30373

so its really a matter of - do we set nothing, and its up to users to set something if they get 429s due to a limit of 0, or do we set something so it works out of the box, but there is a cap that again would require users to set a value that suits their use case? I am ok either way, I don't know what the right way is - either set something (across the board) or set nothing (`null` - again, across the board)

Reply:
Thanks for pointing out the issues. I've encountered this problem in the past, but I didn't realize the reason.

Can we not have a default number here (`null`) but then use `coalesce()` in the code to get the provided value (if specified) or the value from `stage_default_route_settings`? This way, it will be closer to the expected behavior.

Reply:
good point - updated to coalesce these with the stage default route settings and removed defaults from the route section of the variable in https://github.com/terraform-aws-modules/terraform-aws-apigateway-v2/pull/54/commits/6154fc6809bea040ffdec0a8245f999479bc73c1